extensions/resteasy-classic/rest-client-config/runtime/src/main/java/io/quarkus/restclient/config/RestClientsConfig.java

```diff
@@ -371,41 +368,13 @@ interface RestClientConfig {
          * The base URL to use for this service. This property or the `uri` property is considered required, unless
          * the `baseUri` attribute is configured in the `@RegisterRestClient` annotation.
          */
-        Optional<String> url();
-
-        /**
-         * Duplicate mapping of {@link RestClientConfig#url()} to keep a reference of the name used to retrieve the
-         * <code>url</code>. We need this to reload the <code>url</code> configuration in case it contains an expression
-         * to <code>${quarkus.http.port}</code>, which is only set after we load the config.
-         */
-        @ConfigDocIgnore
-        @WithName("url")
-        ConfigValue urlValue();
-
-        default Optional<String> urlReload() {
-            SmallRyeConfig config = ConfigProvider.getConfig().unwrap(SmallRyeConfig.class);
-            return config.getOptionalValue(urlValue().getName(), String.class);
-        }
+        Optional<Address> url();
 
         /**
          * The base URI to use for this service. This property or the `url` property is considered required, unless
          * the `baseUri` attribute is configured in the `@RegisterRestClient` annotation.
          */
-        Optional<String> uri();
-
-        /**
-         * Duplicate mapping of {@link RestClientConfig#uri()} to keep a reference of the name used to retrieve the
-         * <code>uri</code>. We need this to reload the <code>uri</code> configuration in case it contains an expression
-         * to <code>${quarkus.http.port}</code>, which is only set after we load the config.
-         */
-        @ConfigDocIgnore
-        @WithName("uri")
-        ConfigValue uriValue();
-
-        default Optional<String> uriReload() {
-            SmallRyeConfig config = ConfigProvider.getConfig().unwrap(SmallRyeConfig.class);
-            return config.getOptionalValue(uriValue().getName(), String.class);
-        }
+        Optional<Address> uri();
```

Comment:
Again, it can be confusing when using stork, as stork will use an URL or baseURL starting with `stork://<service-name>`.

It's unclear if you need to use Stork or an Address.